lib/linter/linter.js

```diff
@@ -922,6 +922,12 @@ function runRules(sourceCode, configuredRules, ruleMapper, parserOptions, parser
                         if (problem.fix && rule.meta && !rule.meta.fixable) {
                             throw new Error("Fixable rules should export a `meta.fixable` property.");
                         }
+                        if (problem.suggestions && !(rule.meta && rule.meta.hasSuggestions === true)) {
+                            throw new Error("Rules with suggestions should set the `meta.hasSuggestions` property to `true`.");
+                        }
+                        if (rule.meta && rule.meta.docs && typeof rule.meta.docs.suggestion !== "undefined") {
```

Comment:
I don’t think we need this as this property is already ignored. The check on line 925 covers the most important case.

Reply:
I'm worried that if we don't enforce the absence of the old, obsolete property, then we'll end up with the confusing and messy situation where some rules have both the new property (`meta.hasSuggestions`) and the old property (`meta.docs.suggestion`). This will happen because plugin developers performing the migration to ESLint 8 may only see the exception `Rules with suggestions should set the meta.hasSuggestions property to true` and not realize or forget that they're also supposed to cleanup/remove the old property. Furthermore, if some rules still have the old property, then some people/tooling may still try to rely on it.

Disallowing the old property also fits with the spirit of many of the [ESLint 6](https://eslint.org/docs/user-guide/migrating-to-6.0.0#rule-config-validating) and [ESLint 7](https://eslint.org/docs/user-guide/migrating-to-7.0.0#rule-tester-strict) changes to ensure that configurations/tests are validated more strictly and do not contain unknown properties. There's a separate question of whether we should apply stricter to the `meta` object in general (I'm in favor of stricter validation wherever possible and would like to pursue this if there's interest), but at the least I hope we disallow this obviously-outdated property.

Reply:
My point is that it doesn’t matter if people have meta.Docs.suggestions because that value is never used by us. If they are using it for some reason, they can keep using it. 

People can put whatever they want in meta.Docs as it’s just designed to hold documentation-type metadata. I can’t think of a reason why we would want to throw an error for anything in that section. People are free to use stuff in there as they see fit.

Reply:
Maybe we could change the wording a bit?

> Rules with suggestions must set the meta.hasSuggestions property to true. meta.docs.suggestions is ignored by ESLint.

That way it sounds less like you need to remove meta.docs.suggestions and more like, “hey, we aren’t using this value.”